src/Events/QR/QR_Code.php

```diff
@@ -0,0 +1,222 @@
+<?php
+/**
+ * The QR Code class for handling QR code generation and display.
+ *
+ * @since TBD
+ */
+
+namespace TEC\Events\QR;
+
+use TEC\Common\QR\QR;
+use TEC\Events\QR\Routes;
+use Tribe\Utils\Element_Attributes;
+use Tribe__Events__Main as TEC;
+
+/**
+ * Class QR_Code
+ *
+ * @since TBD
+ *
+ * @package TEC\Events\QR
+ */
+class QR_Code {
+	/**
+	 * The QR code instance.
+	 *
+	 * @since TBD
+	 * @var QR
+	 */
+	private $qr_code;
+
+	/**
+	 * The routes instance.
+	 *
+	 * @since TBD
+	 * @var Routes
+	 */
+	private $routes;
+
+	/**
+	 * The QR codes upload directory path.
+	 *
+	 * @since TBD
+	 * @var string
+	 */
+	private $qr_dir;
+
+	/**
+	 * The QR codes upload directory URL.
+	 *
+	 * @since TBD
+	 * @var string
+	 */
+	private $qr_url;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since TBD
+	 */
+	public function __construct() {
+		$this->qr_code = tribe( QR::class );
+		$this->routes  = tribe( Routes::class );
```

Comment:
Here you can also pass those in the constructor instead - if you have our controller retrieve this it will figure out dependencies just fine!